src/west/manifest.py

```diff
@@ -66,6 +66,9 @@
 # though it's just a str in each individual file right now.
 WestCommandsType = Union[str, List[str]]
 
+# Type for the submodules value passed through the manifest file.
+SubmodulesType = List[Dict]
```

Comment:
The Project class is meant to be a high level view, so exposing unparsed manifest dictionary data like this feels a bit out of character for the API to me. Is it possible to make this a list of structured objects, like a `List[submomdule]`, where `submodule` is a `typing.namedtuple` that specifies things about the submodule itself, like its name and path within the superproject?

Reply:
I agree with approach you suggested. I tried to follow it, please verify if current version looks better.